lib/puppet/provider/firewall/iptables.rb

```diff
@@ -452,6 +455,8 @@ def self.rule_to_hash(line, table, counter)
     values = values.gsub(%r{(!\s+)?--tcp-flags (\S*) (\S*)}, '--tcp-flags "\1\2 \3"')
     # --hex-string output is in quotes, need to move ! inside quotes
     values = values.gsub(%r{(!\s+)?--hex-string "(\S*?)"}, '--hex-string "\1\2"')
+    # --condition output is in quotes, need to move ! inside quotes
+    values = values.gsub(%r{(!\s+)?--condition "(\S*?)"}, '--condition "\1\2"')
```

Comment:
You could use `!` to mutate the value of `values`
```suggestion
    values.gsub!(%r{(!\s+)?--condition "(\S*?)"}, '--condition "\1\2"')
```